test/date_types.test.js

```diff
@@ -0,0 +1,261 @@
+// Copyright IBM Corp. 2013,2016. All Rights Reserved.
+// Node module: loopback-connector-mysql
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+require('./init.js');
+var assert = require('assert');
+
+var db, DateModel;
+
+describe('MySQL DATE, DATETTIME, TIMESTAMP types on server with local TZ', function() {
+  var date;
+  var dateOnly = new Date(2015, 10, 25); //2015-12-25
+  var timezone = getTimeZone();
+
+  before(function(done) {
+    prepareModel('local', done);
+  });
+
+  it('should set local timezone in mysql ' + timezone, function(done) {
+    query("SET @@session.time_zone = '" + timezone + "';", function(err) {
+      assert.ok(!err);
+      done();
+    });
+  });
+
+  it('should create a model instance with dates', function(done) {
+    date = new Date();
+    date.setMilliseconds(0);
+    DateModel.create({
+      datetimeField: date,
+      timestampField: date,
+      dateField: dateOnly,
+    }, function(err, obj) {
+      assert.ok(!err);
+      done();
+    });
+  });
+
+  it('should get model instance', function(done) {
+    DateModel.findOne({
+      where: {
+        id: 1,
+      },
+    }, function(err, found) {
+      assert.ok(!err);
+      assert.equal(found.datetimeField.toISOString(), date.toISOString());
+      assert.equal(found.timestampField.toISOString(), date.toISOString());
+      assert.equal(found.dateField.toISOString(), dateOnly.toISOString());
+      done();
+    });
+  });
+
+  it('timestampField shoud equal DEFAULT CURRENT_TIMESTAMP field', function(done) {
+    DateModel.findOne({
+      where: {
+        id: 1,
+      },
+    }, function(err, found) {
+      assert.ok(!err);
+      assert.equal(found.timestampField.toISOString(), found.timestampDefaultField.toISOString());
+      done();
+    });
+  });
+
+  it('should find model instance by datetime field', function(done) {
+    DateModel.findOne({
+      where: {
+        datetimeField: date,
+      },
+    }, function(err, found) {
+      assert.ok(!err);
+      assert.ok(found);
```

Comment:
`assert.equal(found.id, 1);` should be enough - we have to be sure that param of each type is working and found proper recored.